controllers/cart.controller.js

```diff
@@ -0,0 +1,78 @@
+const { Cart } = require('../models/cart.model');
+
+const getAllProductsInCart = async (req, res) => {
+  try {
+    const { userID } = req.user;
+
+    const cart = await Cart.findById(userID).populate('products._id');
+    const cartProducts = cart.products.map((productItem) => {
+      return { ...productItem._id._doc, qty: productItem.qty };
+    });
+
+    const finalCart = { _id: cart._id, products: cartProducts };
+
+    res.json({ success: true, cart: finalCart });
+  } catch (error) {
+    res.json({ success: false, message: 'Error retrieving cart', errorMessage: error.message });
+  }
+};
+
+const addProductInCart = async (req, res) => {
+  try {
+    const { userID } = req.user;
+    let { product } = req.body;
+
+    const user = await Cart.findById(userID);
+    if (!user) {
```

Comment:
i saw in your frontend code that `addToCart` functionality is only for users who are logged in or we can say users which are signed up in your app. If users are signed up then `Cart.findById(userId)` will surely give a result.
So i am unable to understand the need to check whether user exists or not, this is what i inferred from your app.